File: app/src/main/java/com/zxcx/zhizhe/ui/card/card/cardDetails/CardDetailsContract.java
package com.zxcx.zhizhe.ui.card.card.cardDetails;

import com.zxcx.zhizhe.mvpBase.GetPostView;
import com.zxcx.zhizhe.mvpBase.IGetPostPresenter;

public interface CardDetailsContract {

    interface View extends GetPostView<CardDetailsBean,CardDetailsBean> {
        void followSuccess();
        void likeSuccess(CardDetailsBean bean);
        void collectSuccess(CardDetailsBean bean);
    }

    interface Presenter extends IGetPostPresenter<CardDetailsBean,CardDetailsBean> {
        void followSuccess();
        void likeSuccess(CardDetailsBean bean);
        void collectSuccess(CardDetailsBean bean);
    }
}

File: app/src/main/java/com/zxcx/zhizhe/ui/card/cardDetails/CardDetailsContract.kt
package com.zxcx.zhizhe.ui.card.cardDetails

import com.zxcx.zhizhe.mvpBase.GetPostView
import com.zxcx.zhizhe.mvpBase.IGetPostPresenter
import com.zxcx.zhizhe.ui.card.hot.CardBean

interface CardDetailsContract {

	interface View : GetPostView<CardBean, CardBean> {
		fun followSuccess(bean: CardBean)
	}

	interface Presenter : IGetPostPresenter<CardBean, CardBean> {
		fun followSuccess(bean: CardBean)
        fun deleteSuccess()
	}
}

